src/components/ingredient-details/ingredient-details.tsx

```diff
@@ -0,0 +1,39 @@
+import ReactDOM from 'react-dom';
+import styles from './ingredient-details.module.css';
+
+import Modal from '../modal/modal'
+import ModalOverlay from '../modal/modal-overlay'
+import Fact from './fact'
+import type { IngredientShape,  FactNameType } from '../../utils/types.js'
+
+interface IngredientDetailsProps {
+  ingredient: IngredientShape,
+  isShow: boolean,
+  hideDetails: () => void,
+}
+
+const IngredientDetails = (props: IngredientDetailsProps) => {
+  const facts = ['calories', 'proteins', 'fat', 'carbohydrates'].map( 
+    name => (
+      <Fact name={name as FactNameType} value={Number(props.ingredient[name as keyof IngredientShape])} key={name}/>
+    )
+  )
```

Comment:
Можно лучше:
можно избежать лишнего использования `as`, если при создании массива использовать приведение к константному массиву
```suggestion

const NUTRIENTS = ['calories', 'proteins', 'fat', 'carbohydrates'] as const;

const IngredientDetails = (props: IngredientDetailsProps) => {
  const facts = NUTRIENTS.map( 
    name => (
      <Fact name={name} value={Number(props.ingredient[name])} key={name}/>
    )
  )
```